src/app/features/automated-status-update/automated-status-update-config-service.ts

```diff
@@ -134,20 +134,38 @@ export class AutomatedStatusUpdateConfigService {
    * Applies rules to dependent entities based on changes in the provided entity.
    * Also prompts user confirmation and saves updates if any changes were made.
    * @param entity - The source entity whose changes should trigger updates
+   * @param entityBeforeChanges The state of the entity before changes were applied to identify what has changed
    */
   public async applyRulesToDependentEntities(
     entity: Entity,
-    changedFields: any,
+    entityBeforeChanges: Entity,
   ): Promise<void> {
     const affectedEntities: AffectedEntity[] = [];
-    const changedEntries = Object.entries(changedFields);
-    await this.applyFieldMappings(changedEntries, entity, affectedEntities);
+    const changedFields = this.getChangedFields(entity, entityBeforeChanges);
+    await this.applyFieldMappings(changedFields, entity, affectedEntities);
 
     if (affectedEntities.length > 0) {
       await this.confirmAndSaveAffectedEntities(affectedEntities);
     }
   }
 
+  private getChangedFields(
+    newEntity: Entity,
+    originalEntity: Entity,
+  ): [string, any][] {
```

Comment:
```suggestion
  ): [string, any][] {
```
I was thinking we could use a more specific type, but I guess we’re using any because the value can be a string, an object, or maybe an address config?